components/script/dom/bindings/str.rs

```diff
@@ -306,11 +306,15 @@ impl DOMString {
     /// A valid number is the same as what rust considers to be valid,
     /// except for +1., NaN, and Infinity.
     /// https://html.spec.whatwg.org/multipage/#valid-floating-point-number
-    pub fn is_valid_number_string(&self) -> bool {
-        let input = &self.0;
-        input.parse::<f64>().ok().map_or(false, |val| {
-            !(val.is_infinite() || val.is_nan() || input.ends_with(".") || input.starts_with("+"))
-        })
+    pub fn is_valid_float_number_string(&self) -> bool {
```

Comment:
I find the terminology here very odd. We usually call them "floats" or "floating point numbers", but never "float number".